src/collision.cpp

```diff
@@ -54,6 +54,22 @@ struct NearbyCollisionInfo {
 	aabb3f box;
 };
 
+// Helper functions:
+// Truncate floating point numbers to specified number of decimal places
+// in order to move all the floating point error to one side of the correct value
+static inline f32 truncate(const f32 val, const u16 exp)
+{
+	return truncf(val * 10.0f * exp) / (10.0f * exp);
```

Comment:
`exp` is misleading. `factor` would fit better to express the linearity.
If you'd instead like to specify the decimal places directly: `pow(10.0f, exp)`
In terms of performance I'd prefer keeping this as a simple factor, and passing `100.0f` as argument below.

Reply:
Not a fan of that solution, you'd be able to pass just any float there, but I'm going to change it if you insist.

Reply:
I don't mind. Regarding performance, a factor would be a better choice. Feel free to use `exp` if you prefer to keep the decimal point argument.

Reply:
Shouldn't matter that much performance wise, the operation should on average be done 3 * #movingboxes / step, but if precision is increased to 64 then this function should be used more often, so
changed as requested.